spring-web/src/main/java/org/springframework/web/util/UriComponentsBuilder.java

```diff
@@ -1023,18 +1023,11 @@ public PathComponent build() {
 			if (this.path.length() == 0) {
 				return null;
 			}
-			String path = this.path.toString();
-			while (true) {
-				int index = path.indexOf("//");
-				if (index == -1) {
-					break;
-				}
-				path = path.substring(0, index) + path.substring(index + 1);
-			}
+			String path = UrlPathHelper.getSanitizedPath(this.path);
```

Comment:
I would prefer to avoid any dependency from `UriComponentsBuilder` on `UrlPathHelper`. The latter is more of a Servlet API specific component while the former is more general. Even if the re-use in this case doesn't touch on anything Servlet specific, it seems very minor and not worth the gain.

Reply:
Good catch, Rossen. I'd rather accept some minor duplication there as well.

Reply:
@rstoyanchev should I move the method into a separate utility class or keep duplicated code fragment?

Reply:
I'd rather keep the duplicated code fragment, unless there is an existing common place to add this to, @rstoyanchev?

BTW a minor thing: The `public void removeTrailingSlash() ` method was intentionally public. We tend to declare API methods as public if they're called from outside even for inner classes, along the lines of "what if this inner class was an externally declared class, would it have to expose this as public method then".

Reply:
Yes, the minor duplication is the preferred outcome.

Reply:
Done